engine/schema/src/main/java/com/cloud/upgrade/SystemVmTemplateRegistration.java

```diff
@@ -0,0 +1,769 @@
+                metaFile = new File(filePath);
+                templatePath = AbsolutetemplatesPath;
+                LOGGER.debug("Looking for file [" + filePath + "] in the classpath.");
+                if (!metaFile.exists()) {
+                    String errMsg = String.format("Unable to locate metadata file in your setup at %s", filePath.toString());
+                    LOGGER.error(errMsg);
+                    throw new CloudRuntimeException(errMsg);
+                }
+            }
+        return templatePath;
+    }
+
+    private static String getHypervisorName(String name) {
+        if (name.equals("xenserver")) {
+            return "xen";
+        }
+        if (name.equals("ovm3")) {
+            return "ovm";
+        }
+        return name;
+
+    }
+
+    private static Hypervisor.HypervisorType getHypervisorType(String hypervisor) {
+        if (hypervisor.equalsIgnoreCase("xen")) {
+            hypervisor = "xenserver";
+        } else if (hypervisor.equalsIgnoreCase("ovm")) {
+            hypervisor = "ovm3";
+        }
+        return Hypervisor.HypervisorType.getType(hypervisor);
+    }
+
+    private static List<Long> getEligibleZoneIds(Connection conn) {
+        List<Long> zones = new ArrayList<Long>();
+        try {
+            PreparedStatement pstmt = conn.prepareStatement(FETCH_DISTINCT_ELIGIBLE_ZONES);
+            ResultSet rs = pstmt.executeQuery();
+            while (rs.next()) {
+                zones.add(rs.getLong(1));
+            }
+        } catch (SQLException e) {
+            String errMsg = "Failed to fetch eligible zones for SystemVM template registration due to: %s";
+            LOGGER.error(String.format(errMsg, e.getMessage()));
+            throw new CloudRuntimeException("Failed to fetch eligible zones for SystemVM template registration");
+        }
+        return zones;
+    }
+
+    private static Pair<String, Long> getNfsStoreInZone(Connection conn, Long zoneId) {
+        String url = null;
+        Long storeId = null;
+        try {
+            PreparedStatement pstmt = conn.prepareStatement(FETCH_IMAGE_STORE_PER_ZONE);
+            if(pstmt != null) {
+                pstmt.setLong(1, zoneId);
+                ResultSet resultSet = pstmt.executeQuery();
+                while (resultSet.next()) {
+                    url = resultSet.getString(1);
+                    storeId = resultSet.getLong(2);
+                }
+            }
+        } catch (SQLException e) {
+            String errMsg = String.format("Failed to fetch NFS store in zone = %s for SystemVM template registration", zoneId);
+            LOGGER.error(errMsg + String.format("due to: %s", e.getMessage()));
+            throw new CloudRuntimeException(errMsg);
+        }
+        return new Pair<>(url, storeId);
+    }
+
+    public static void mountStore(String storeUrl) {
+        try {
+            if (storeUrl != null) {
+                String path = storeUrl.split("://")[1];
```

Comment:
Can we use URI/Url to get scheme path etc? https://docs.oracle.com/javase/7/docs/api/java/net/URI.html or similar?

Reply:
Addressed with https://github.com/apache/cloudstack/pull/4329/commits/c91c39652d301c9bfc9dfd227531e3b262d87218#diff-a9c9a38684718059c060de404bf9529de96e502f1e81b30793e6a32f725042a9L409-R412